src/core/SkillMatrix/service/SkillMatrixService.ts

```diff
@@ -78,4 +78,13 @@ export class SkillMatrixService {
   async delete(uid: string) {
     await this.skillMatrixRepository.delete(uid)
   }
+
+  async deleteSkillMatrixByCompany(companyDomain: string): Promise<void> {
+    const allUsersCompany = await this.userProfileService.getByCompany(companyDomain)
```

Comment:
Se dovesse esserci un errore ad un certo punto del ciclo for la cancellazione degll'entità "SkillMatrix" verrebbe interrotta lasciando il DB in stato inconsistente